### PR TITLE
Add timers to seq_example

### PR DESCRIPTION
**Pull request by SylvainJoube**
I have added timers to the CPU full-chain `seq_example.cpp`. If you think this is a good idea, I will also add timers to the `seeding_example.cpp` file. I think it would be quite interesting to have a record for algorithm performance, both in terms of efficiency visualized by ROOT, but also in terms of execution time.

I understand that this may be quite complicated to set up, since it would require running the timed benchmarks on the same machine and with the same input data, but I think it would be very interesting to see the evolution of `traccc` performance over time. This may not be a priority, since (as far I understand it) `traccc` is primarily aimed at demonstrating the feasibility of having a common code base for the CPU + any compute accelerator, with portable performance (as much as possible).

Here is an example of output I got:

```
$ ${traccc_build_folder}/bin/traccc_seq_example --detector-file=geometries/odd/odd_geometry_detray.json --use-detray-detector --digitization-file=geometries/odd/odd-digi-geometric-config.json --input-directory=odd/muon100GeV-geant4/ --input-events=2

Running Full Tracking Chain on the Host

>>> Detector Options <<<
  Detector file       : geometries/odd/odd_geometry_detray.json
  Material file       : 
  Surface rid file    : 
  Use detray::detector: yes
  Digitization file   : geometries/odd/odd-digi-geometric-config.json
>>> Input Data Options <<<
  Input data format             : csv
  Input directory               : odd/muon100GeV-geant4/
  Number of input events        : 2
  Number of input events to skip: 0
>>> Clusterization Options <<<
  Target cells per partition: 1024
>>> Track Seeding Options <<<
  None
>>> Track Finding Options <<<
  Track candidates range   : 3:100
  Maximum Chi2             : 30
  Maximum branches per step: 4294967295
>>> Track Propagation Options <<<
  Constraint step size : 3.40282e+38 [mm]
  Overstep tolerance   : -100 [um]
  Mask tolerance       : 15 [um]
  Search window        : 0 x 0
  Runge-Kutta tolerance: 0.0001
>>> Track Ambiguity Resolution Options <<<
  Run ambiguity resolution : yes
>>> Performance Measurement Options <<<
  Run performance checks: no

WARNING: No material in detector
WARNING: No entries in volume finder
Detector check: OK
WARNING: No material in detector
WARNING: No entries in volume finder
Detector check: OK
@greedy_ambiguity_resolution_algorithm: Iteration_count: 6416
@greedy_ambiguity_resolution_algorithm: Checking result validity...
@greedy_ambiguity_resolution_algorithm: OK 1/2: every removed track had at least one commun measurement with another track.
@greedy_ambiguity_resolution_algorithm: OK 2/2: each selected_track shares at most (_config.maximum_shared_hits - 1)(=0) measurement(s)
@greedy_ambiguity_resolution_algorithm: state.selected_tracks.size() = 1729
@greedy_ambiguity_resolution_algorithm: Iteration_count: 5869
@greedy_ambiguity_resolution_algorithm: Checking result validity...
@greedy_ambiguity_resolution_algorithm: OK 1/2: every removed track had at least one commun measurement with another track.
@greedy_ambiguity_resolution_algorithm: OK 2/2: each selected_track shares at most (_config.maximum_shared_hits - 1)(=0) measurement(s)
@greedy_ambiguity_resolution_algorithm: state.selected_tracks.size() = 1752
==> Statistics ... 
- read     163156 cells from 22437 modules
- created  63510 measurements. 
- created  63510 space points. 
- created  11806 seeds
- found    15766 tracks
- fitted   15766 tracks
- resolved 3481 tracks
==> Elapsed times...
                    Read cells  403 ms
                Clusterization  10 ms
          Spacepoint formation  1 ms
                       Seeding  205 ms
       Track params estimation  3 ms
                 Track finding  6035 ms
                 Track fitting  2108 ms
    Track ambiguity resolution  1655 ms
                     Wall time  10423 ms
```

### CODE REVIEW

File: examples/run/cpu/seq_example.cpp [hunk 1]
```diff
@@ -55,6 +56,14 @@
 #include <map>
 #include <memory>
 
+// Timer macros
+// I used a pointer to avoid indenting the whole code and make it more readable.
+#define TIMER_START(unique_var_name, str_name) \
+    auto unique_var_name =                     \
+        new traccc::performance::timer(str_name, elapsedTimes);
+
+#define TIMER_STOP(unique_var_name) delete unique_var_name;
```
Comment (krasznaa) [inline examples/run/cpu/seq_example.cpp:65]: This is not the way...

`traccc::performance::timer` objects are meant to be created on the stack, in appropriate contexts. Adding extra code blocks where needed. I.e.

```c++
{
    traccc::performance::timer timer{"Read cells", elapsedTimes};
    traccc::io::cell_reader_output readOut(&host_mr);
}
```

image

**Review by krasznaa — CHANGES_REQUESTED** [review]
I'm very happy to add timing measurements to this executable. But let's not use macros / smart pointers for it.

File: examples/run/cpu/seq_example.cpp (hunk 1 above)
Comment (SylvainJoube) [reply examples/run/cpu/seq_example.cpp]: Thank you for your review! I removed the macros / heap allocations and added extra code blocks where needed. I had to declare every `...algorithm::output_type` at the beginning of each event iteration.

I hope these changes will not affect readability too much!

**Review by krasznaa — APPROVED** [review]
I like it. Just please resolve whatever conflict came up in the file. And them I'm good with this. :wink:

**Comment by SylvainJoube** [comment]
Thanks! I rebased the branch 😺

**Pull request merged**